rviz_default_plugins/src/rviz_default_plugins/robot/robot_joint.cpp

```diff
@@ -265,40 +277,39 @@ void RobotJoint::calculateJointCheckboxesRecursive(
   }
 }
 
-void RobotJoint::getChildLinkState(
-  int & links_with_geom,
-  int & links_with_geom_checked,
-  int & links_with_geom_unchecked,
-  bool recursive) const
+RobotLink * RobotJoint::links_checked_and_unchecked(
+  int & links_with_geom_checked, int & links_with_geom_unchecked)
 {
-  links_with_geom = 0;
   links_with_geom_checked = 0;
   links_with_geom_unchecked = 0;
 
   RobotLink * link = robot_->getLink(child_link_name_);
-  if (link == nullptr) {
-    return;
+  if (!link) {
+    return nullptr;
   }
   if (link->hasGeometry()) {
     bool checked = link->getLinkProperty()->getValue().toBool();
     links_with_geom_checked += checked ? 1 : 0;
     links_with_geom_unchecked += checked ? 0 : 1;
   }
+  return link;
+}
+
+void RobotJoint::getChildLinkState(
+  int & links_with_geom,
+  int & links_with_geom_checked,
+  int & links_with_geom_unchecked,
+  bool recursive)
+{
+  links_with_geom = 0;
```

Comment:
If the call to links_checked_and_unchecked returned a nullptr and this method returned without making the call to links_with_geom, the output variable links_with_geom would have been initialized to 0. It seems a bit cleaner to me to also initialize the other two as well. I understand that if links_with_geom is zero the other two are invalid, but still, I think not allowing for the possibility of error on the caller side (for example, not initializing the input variables, calling this method, it fails, and then the caller uses links_with_geom_checked (or unchecked) without first checking links_with geom). I mention this because I saw a very similar situation in some X11 code that I was debugging because of a failure.